src/test/integration/services/question.spec.js

```diff
@@ -0,0 +1,233 @@
+    expect(response.updatedAt).toBeInstanceOf(Date)
+
+    question = response
+    questionMatchObject = {
+      title: question.title,
+      text: question.text,
+      answers: [{ text: question.answers[0].text, isCorrect: question.answers[0].isCorrect }],
+      type: question.type,
+      category: question.category._id,
+      author: question.author,
+      resourceType: question.resourceType,
+      createdAt: question.createdAt,
+      updatedAt: question.updatedAt
+    }
+  })
+
+  it('Should find question by Id', async () => {
+    const response = await questionService.getQuestionById(question._id)
+    expect(response).toMatchObject(questionMatchObject)
+  })
+
+  it('Should find question by author', async () => {
+    const matchOption = {
+      author: question.author.toString()
+    }
+    const sortOption = {
+      updatedAt: 'asc'
+    }
+
+    const response = await questionService.getQuestions(matchOption, sortOption)
+    expect(response.count).toEqual(1)
+    expect(response.items[0]).toMatchObject(questionMatchObject)
+  })
+
+  it('Should find question by title', async () => {
+    const matchOption = {
+      title: 'test'
+    }
+    const sortOption = {
+      updatedAt: 'asc'
+    }
+
+    const response = await questionService.getQuestions(matchOption, sortOption)
+    expect(response.count).toEqual(1)
+    expect(response.items[0]).toMatchObject(questionMatchObject)
+  })
+
+  it('Should find question by category', async () => {
+    const matchOption = {
+      category: category._id
+    }
+    const sortOption = {
+      updatedAt: 'asc'
+    }
+
+    const response = await questionService.getQuestions(matchOption, sortOption)
+    expect(response.count).toEqual(1)
+    expect(response.items[0]).toMatchObject(questionMatchObject)
+  })
+
+  it('Should sort desc by updateData', async () => {
+    const mockData = {
+      author: user['_id'].toString(),
+      data: {
+        title: 'test2',
+        text: 'test',
+        answers: [
+          {
+            text: 'test',
+            isCorrect: true
+          }
+        ],
+        type: enums.QUESTION_TYPE_ENUM[0],
+        category: category['_id'].toString()
+      }
+    }
+    await questionService.createQuestion(mockData.author, mockData.data)
+
+    const matchOption = {
+      category: category._id
+    }
+    const sortOption = {
+      updatedAt: 'desc'
+    }
+
+    const response = await questionService.getQuestions(matchOption, sortOption)
+    expect(response.items[1]).toMatchObject(questionMatchObject)
+  })
+
+  it("Should not update question when currentUser doesn't match author", async () => {
+    const mockData = {
+      id: question._id.toString(),
+      toUpdate: {
+        title: 'updateTest'
+      },
+      userId: '12345678'
+    }
+
+    expect(
+      async () => await questionService.updateQuestion(mockData.id, mockData.userId, mockData.toUpdate)
+    ).rejects.toThrow(createForbiddenError())

```

Comment:
In integration test we need to check the final result. In our case we can check that DB was not updated. For example maybe error generated but this update already is done.

Reply:
added verifacation that service don't call db update action

Reply:
Can you also add `expect(questionMock.save).toHaveBeenCalledWith` in other test cases?